### PR TITLE
Confidential bundle store demo

### PR DESCRIPTION
**Pull request by Ruteri**
Adds an example bundle confidential store contract as well as a demo webapp for it.
Demo can be (currently) viewed at https://demo.sirrah.suave.flashbots.net

### CODE REVIEW

File: src/KeyHelper.sol [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    address owner;
+
+    constructor(KeyManager_v0 _keymgr) {
+        keymgr = _keymgr; // Note that we only derive privkey, so we don't require derivekey be called with the address of this (private) contract. The key manager has to be bootstrapped.
+        owner = msg.sender; // Can we really trust the sender though? The kettle could be spoofing it
+    }
+
+    function hashSecret() public returns (bytes32) {
+        // We deploy a different contract as the derivePriv will be different in this scenario!
+        require(msg.sender == owner);
```
Comment (MoeMahhouk) [inline src/KeyHelper.sol:16]: wouldn't this restrict it to only one owner?
Would that case an issue for the case of several kettles onboarded and share the keymgr deriveKey?

Comment (Ruteri) [reply src/KeyHelper.sol:16]: I think in this case the sender in this case will be the parent (onchain) contract, which is available to everyone

File: src/KeyHelper.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    function encrypt(string memory message) internal view returns (string memory) {
+        return string(encrypt(abi.encodePacked(message), keymgr.Suave().localRandom()));
+    }
+
+    function decrypt(bytes memory ciphertext) internal returns (bytes memory message) {
+        // Note that it's possible to decrypt an auth_encrypted message
+        return PKE.decrypt(privkey(), ciphertext);
+    }
+
+    function auth_encrypt(bytes memory message, bytes32 r) internal returns (bytes memory) {
```
Comment (amiller) [inline src/KeyHelper.sol:66]: This looks good. It's using keccak256 as a (non-standard) MAC. keccak256 shouldn't be vulnerable to length extension so this is reasonable

**Comment by Ruteri** [comment]
Solves https://github.com/flashbots/andromeda-sirrah-contracts/issues/33

**Pull request merged**